packages/blocks/src/types/block-attributes.ts

```diff
@@ -0,0 +1,319 @@
+/**
+ * Block Attribute Interfaces
+ *
+ * @module BlockAttributes
+ */
+
+/**
+ * External dependencies
+ */
+import type { ReactChild } from 'react';
+
+/**
+ * Types of data stored by block attributes.
+ *
+ * @see {@link https://developer.wordpress.org/block-editor/reference-guides/block-api/block-attributes/#type-validation Type Validation} on developer.wordpress.org
+ */
+export type AttributeSchemaType =
+	| 'string'
+	| 'boolean'
+	| 'object'
+	| 'null'
+	| 'array'
+	| 'integer'
+	| 'number';
+
+export type BaseAttributeSource = {
+	type: AttributeSchemaType | AttributeSchemaType[];
+};
+
+/**
+ * Used in [get-block-attributes.js](../api/parser/get-block-attributes.js)
+ *
+ * - [ ] TODO this isn't defined in {@link https://github.com/WordPress/gutenberg/blob/trunk/schemas/json/block.json block.json}
+ * - [ ] TODO is this only used internally?
+ *
+ * @internal
+ */
+export interface Raw {
+	source: 'raw';
+	type: 'string'; // TODO is this correct?
+}
```

Comment:
"Raw" seems a bit broad, so I wasn't quite sure what it intends. We should likely prefix it with something, since this type is exported and may mean different things in different contexts.